x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -286,6 +292,7 @@ export const AlertsTableComponent: FC<DetectionEngineAlertTableProps> = ({
         showSortSelector: !isEventRenderedView,
       },
       dynamicRowHeight: isEventRenderedView,
+      customRefetch,
```

Comment:
Do you think, instead of passing a new prop to trigger actions table, we can make use of `onUpdate` prop?

We can may be trigger `customRefecth` everytime the table is updated? But I think there might be cases which it will get into infinite render loop.


Alternatively, I would ask you to take a look at below file. It creates bulk actions and passes to the Alert Table. May be here we can call `global query refetch` directly. This might keep `refetch` responsibilities well separated and we will probably avoid unintended re-renders.

https://github.com/elastic/kibana/blob/b0f8ee7f5e5a755e03723f79e09badbcae374dc0/x-pack/plugins/security_solution/public/detections/hooks/trigger_actions_alert_table/use_bulk_actions.tsx#L59-L65

Let me know what you think.

Reply:
@logeekal thank you for pointing to this file! i have moved the logic to the alert refetch, which should further limit when the query refresh happens.

https://github.com/elastic/kibana/pull/183674/files#diff-4c600251369f5ce9a45d5cc4d74deca0d2507ecc2170a85e08b48dad40d5a298L85-L93